app/assets/javascripts/modules/track-qa-choices.js

```diff
@@ -0,0 +1,83 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (global, GOVUK) {
+  'use strict'
+
+  GOVUK.Modules.TrackQaChoices = function () {
+    this.start = function (element) {
+      track(element[0])
+    }
+
+    function getQuestionKey(submittedForm) {
+      return submittedForm.getAttribute('data-question-key');
+    }
+
+    function triggerEventForMultipleChoice(submittedForm) {
+      var checkedOptions = submittedForm.querySelectorAll('input:checked')
+      var questionKey = getQuestionKey(submittedForm)
+
+      if (checkedOptions.length) {
+        checkedOptions.forEach(function (checkedOption) {
+          var checkedOptionId = checkedOption.getAttribute('id')
+          var checkedOptionLabel = document.querySelectorAll('label[for="' + checkedOptionId + '"]')[0].innerText.trim()
+          var eventLabel = checkedOptionLabel.length
+            ? checkedOptionLabel
+            : checkedOption.value
+
+          var options = { transport: 'beacon', label: eventLabel }
+
+          GOVUK.analytics.trackEvent('question_answer', questionKey, options)
+        })
+      } else {
+        // Skipped questions
+        var options = { transport: 'beacon', label: 'no choice' }
+
+        GOVUK.analytics.trackEvent('question_answer', questionKey, options)
+      }
+    }
+
+    function triggerEventForSelectOption(submittedForm) {
+      var selectInputs = submittedForm.querySelectorAll('select')
+      var questionKey = getQuestionKey(submittedForm)
+
+      if (selectInputs.length) {
+        selectInputs.forEach(function (select) {
+          var value = select.value
+          var label = select.options[select.selectedIndex].innerHTML
+          var eventLabel = label.length ? label : value
+
+          var options = { transport: 'beacon', label: eventLabel }
+
+          GOVUK.analytics.trackEvent('question_answer', questionKey, options)
+        })
+      } else {
+        // Skipped questions
+        var options = { transport: 'beacon', label: 'no choice' }
+
+        GOVUK.analytics.trackEvent('question_answer', questionKey, options)
+      }
+    }
+
+    function track (element) {
+      element.addEventListener('submit', function (event) {
+        var submittedForm = event.target
+        var questionType = submittedForm.getAttribute('data-type')
+
+        switch (questionType) {
+          case 'checkbox_question':
+          case 'multiple_choice_question':
+              triggerEventForMultipleChoice(submittedForm);
+            break;
+          
+          case 'country_select_question':
+            triggerEventForSelectOption(submittedForm);
+            break;
+          
+          default:
+            break;
```

Comment:
How come we aren't supporting other question types?

Reply:
We are only tracking pre-defined responses at the moment as we have no way of stripping out PII